### PR TITLE
Decompile Unk_0255418.s

### PR DESCRIPTION
**Pull request by adrienntindall**
Time functions, daily resets, and apricorn tree functions;

### CODE REVIEW

File: src/unk_02055418.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    sysinfo->unk2C = GF_RTC_DateTimeToSec();
+}
+
+BOOL sub_02055670(FieldSystem* fsys) {
+    SYSINFO_RTC* sysinfo = Sav2_SysInfo_RTC_get(fsys->savedata);
+    return sub_02028E1C(sysinfo);
+}
+
+UnkStruct_020556FC* sub_02055680(UnkStruct_020556FC* unkA, u32 unkB) {
+    UnkStruct_020556FC* unkPtr = AllocFromHeap(unkB, sizeof(UnkStruct_020556FC));
```
Comment (tgsm) [inline src/unk_02055418.c:121]: `unkB` is a HeapID

Comment (adrienntindall) [reply src/unk_02055418.c]: Fixed! Thanks for the catch!

**Review by tgsm — CHANGES_REQUESTED** [review]
There's also a lot of implicit declarations here

File: include/unk_02014DA0.h [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #define POKEHEARTGOLD_UNK_02014DA0_H
 
 #include "unk_02022D74.h"
-#include "spl.h"
+#include "library/spl.h"
```
Comment (PikalaxALT) [inline include/unk_02014DA0.h:5]: Revert this

File: src/unk_02055418.c [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+#include "script.h"
+#include "sys_vars.h"
+#include "constants/sprites.h"
+#include "unk_02055418.h"
+
+void FieldSys_StartBugContestTimer(FieldSystem* fsys) {
+    RTCDate date;
+    RTCTime time;
+    SYSINFO_RTC* sysinfo = Sav2_SysInfo_RTC_get(fsys->savedata); 
+    if (!sysinfo->initialized) return;
```
Comment (PikalaxALT) [inline src/unk_02055418.c:14]:
```suggestion
    if (!sysinfo->initialized) {
        return;
    }
```

File: src/unk_02055418.c [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        sysinfo->date = *date;
+        sysinfo->time = *time;
+        return;
+    }
+    s32 delta_seconds = seconds - sys_seconds;
+    if (delta_seconds > 0) {
+        sub_02092F30(sub_02092DEC(fsys), seconds);
+    }    
+    s32 minutes = delta_seconds / 60;
+    if (minutes <= 0) return;
```
Comment (PikalaxALT) [inline src/unk_02055418.c:46]:
```suggestion
    if (minutes <= 0) {
        return;
    }
```

File: src/unk_02055418.c [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    if (id != SPRITE_BONGURI_R && 
+        id != SPRITE_BONGURI_Y && 
+        id != SPRITE_BONGURI_B && 
+        id != SPRITE_BONGURI_G && 
+        id != SPRITE_BONGURI_P && 
+        id != SPRITE_BONGURI_W && 
+        id != SPRITE_BONGURI_BK) {
+        return FALSE;
+    }
+    if (sub_02055780(fsys, mapObject) != TRUE) return FALSE;
```
Comment (PikalaxALT) [inline src/unk_02055418.c:161]:
```suggestion
    if (sub_02055780(fsys, mapObject) != TRUE) {
        return FALSE;
    }
```

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

**Review by PikalaxALT — APPROVED** [review]

**Review by tgsm — APPROVED** [review]

**Pull request merged**